day_4_camp_cleanup/day_4b.rb

```diff
@@ -0,0 +1,57 @@
+require 'minitest/autorun'
+require 'set'
+
+class CampCleanup
+
+  def initialize(elf_pairs)
+    @elf_pairs = elf_pairs
+  end
+
+  def number_of_overlapping_pairs
+    elf_pairs.inject(0) do |sum, elf_pair|
+      sum += 1 if overlapping_pairs?(elf_pair)
+      sum
+    end
+  end
+
+private
+
+  attr_reader :elf_pairs
+
+  def overlapping_pairs?(elf_pair)
+    overlap = first_elfs_section(elf_pair) & second_elfs_section(elf_pair)
+
+    return true if overlap.any?
+    false
```

Comment:
Nitpicking, but you don't need this line.

If `overlap.any?` is `false`, that line is returning `false`, and thus, the whole method returns `false`.

Reply:
Great comment. Will remove that line.